src/NLog/LogFactory.cs

```diff
@@ -254,8 +254,19 @@ public LoggingConfiguration Configuration
                     if (oldConfig != null)
                     {
                         InternalLogger.Info("Closing old configuration.");
-                        Flush();
-                        oldConfig.Close();
+                        if (!FlushAllTargetsSync(oldConfig, DefaultFlushTimeout, ThrowExceptions))
+                        {
+                            var manualResetEvent = new ManualResetEvent(false);
+                            AsyncHelpers.ForEachItemInParallel(new[] { oldConfig }, ex => manualResetEvent.Set(), (old, cont) => { old.Close(); cont(null); });
```

Comment:
Maybe we could make a new method in AsyncHelpers? `ForEachItemInParallel(new[] { oldConfig }` is a bit strange to read. 

(PS I prefer also to break this line with some newlines, e.g. something like this:

```c#
 AsyncHelpers.ForEachItemInParallel(new[] { oldConfig },
     ex => manualResetEvent.Set(), 
     (old, cont) => { old.Close(); cont(null); }
   );

```

Reply:
Added newlines.